pctasks/run/pctasks/run/workflow/executor/remote.py

```diff
@@ -557,53 +577,22 @@ def complete_job_partitions(
                                     )
+                                completed_job_count += 1
+                                logger.info(
+                                    f"Job partition completed: {job_id}:{part_id}"
+                                )
 
-                                # Handle job completion
-                                if (
-                                    job_part_state.current_task is None
-                                    and job_part_state.status
-                                    != JobPartitionStateStatus.FAILED
-                                ):
-                                    try:
-                                        job_part_state.status = (
-                                            JobPartitionStateStatus.SUCCEEDED
-                                        )
-                                        update_job_partition_run_status(
-                                            container,
-                                            run_id=run_id,
-                                            job_partition_run_id=part_run_record_id,
-                                            status=JobPartitionRunStatus.COMPLETED,
-                                        )
-                                        self.handle_job_part_notifications(
-                                            job_part_state
-                                        )
-                                    except Exception:
-                                        job_part_state.status = (
-                                            JobPartitionStateStatus.FAILED
-                                        )
-                                        failed_job_count += 1
-                                        update_job_partition_run_status(
-                                            container,
-                                            run_id=run_id,
-                                            job_partition_run_id=part_run_record_id,
-                                            status=JobPartitionRunStatus.FAILED,
-                                        )
-                                    completed_job_count += 1
-                                    logger.info(
-                                        f"Job partition completed: {job_id}:{part_id}"
-                                    )
-
-                                _report_status()
+                            _report_status()
 
-                    time.sleep(0.25 + ((random.randint(0, 10) / 100) - 0.05))
+                time.sleep(0.25 + ((random.randint(0, 10) / 100) - 0.05))
 
-                logger.info(f"Partition group {group_id} completed!")
+            logger.info(f"Partition group {group_id} completed!")
 
-            except Exception as e:
-                logger.exception(e)
-                raise
+        except Exception as e:
+            logger.exception(e)
+            raise
```

Comment:
Slight nitpick: but these

```
except Exception as e:
logger.exception(e)
raise
```

might be better handled with a [`sys.excepthook`](https://docs.python.org/3/library/sys.html#sys.excepthook) at the application level (I assume this is where in pctasks server or something like that). I also don't know what the caller(s) exception handling looks like.

Using `sys.excepthook` means we only need to worry about otherwise unhandled exceptions in one spot.

Reply:
This is in the workflow executor.

I'm not sure how sys.excepthook works in a threaded context - I'm assuming the thread pool catches the exception so that it is returned as the future.exception() - which would mean sys.excepthook might not come into play there? Either way, I think I'd prefer being explicit about where the exception was logged rather than rely on the caller not handling it/handling it and logging it/reasoning about exception handling in threads.